src/utils/url_utils/src/url_utils.c

```diff
@@ -0,0 +1,70 @@
+/**
+ * @file url_utils.c
+ * @brief Implements the url utils.
+ *
+ * @copyright Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+#include "aduc/url_utils.h"
+#include "aduc/http_url.h"
+#include "aduc/string_c_utils.h"
+
+ADUC_Result ADUC_UrlUtils_GetLastPathSegmentOfUrl(const char* url, STRING_HANDLE* outLastPathSegment)
+{
+    ADUC_Result result = { .ResultCode = ADUC_GeneralResult_Failure, .ExtendedResultCode = 0 };
+
+    STRING_HANDLE tmp = NULL;
```

Comment:
*If* we not removing libCurl, may be using libCurl will be easier here?

Reply:
yes, it was better and easier.  also using strrchr